pkg/ext-proc/backend/provider.go

```diff
@@ -217,3 +227,40 @@ func (p *Provider) refreshMetricsOnce() error {
 	}
 	return errs
 }
+
+func (p *Provider) flushPrometheusMetricsOnce() {
+	klog.V(logutil.DEFAULT).Infof("Flushing Prometheus Metrics")
+
+	pool, _ := p.datastore.getInferencePool()
+	if pool == nil {
+		// No inference pool or not initialize.
+		return
+	}
+
+	var kvCacheTotal float64
+	var queueTotal int
+
+	podTotalCount := 0
+	pods, err := p.datastore.getPods()
```

Comment:
just iterate over the entries in the podMetrics map directly, the pods in the datastore will be removed.

Reply:
Done.